noxfile.py

```diff
@@ -36,11 +37,19 @@ def set_environment_variables(env_dict, session):
 
 @nox.session(name="pybamm-requires")
 def run_pybamm_requires(session):
-    """Download, compile, and install the build-time requirements for Linux and macOS: the SuiteSparse and SUNDIALS libraries."""  # noqa: E501
-    set_environment_variables(PYBAMM_ENV, session=session)
+    """Download, compile, and install the build-time requirements for Linux and macOS"""
+    force_rebuild = "--force" in session.posargs
+    sundials_path = Path("/path/to/sundials")
+    suitesparse_path = Path("/path/to/suitesparse")
```

Comment:
The relevant files whose existence is to be checked for each of the build-time components are these shared objects, i.e., `.so` files (these will be, similarly, dynamic libraries in `.dylib` file format on macOS):

<details><summary>SUNDIALS</summary>
<p>

```bash
libsundials_idas.so
libsundials_sunlinsolklu.so
libsundials_sunlinsoldense.so
libsundials_sunlinsolspbcgs.so
libsundials_sunlinsollapackdense.so
libsundials_sunmatrixsparse.so
libsundials_nvecserial.so
libsundials_nvecopenmp.so
```

</p>
</details> 

<details><summary>SuiteSparse</summary>
<p>

```
libsuitesparseconfig.so
```

</p>
</details>

<details><summary>For the KLU component</summary>
<p>

```
libsuitesparseconfig.so
libklu.so
libamd.so
libcolamd.so
libbtf.so
```

</p>
</details>

If these files are not found, we should run the `install_KLU_Sundials.py` script

For more information: the SuiteSparse KLU module depends on these four components: suitesparseconfig, AMD, COLAMD, and BTF.